lib/tasks/user.rake

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require_relative 'task_helpers'
+
+namespace :user do
+  desc 'Create a user'
+  task :create, %i[name email password role] => :environment do |_task, args|
+    # Default values.
+    user = {
+      provider: 'greenlight',
+      verified: true,
+      status: :active,
+      language: I18n.default_locale
+    }.merge(args)
+
+    user[:role] = SettingGetter.new(setting_name: 'DefaultRole', provider: 'greenlight').call if user[:role].blank?
+
+    check_role!(user:)
+    user = User.new(user)
+
+    display_user_errors(user:) unless user.save
+
+    success 'User account was created successfully!'
+    info "  Name: #{user.name}"
+    info "  Email: #{user.email}"
+    info "  Password: #{user.password}"
+    info "  Role: #{user.role.name}"
+    info "  Verified: #{user.verified}"
+    info "  Status: #{user.status}"
+    info "  Language: #{user.language}"
+
+    exit 0
+  end
+
+  private
+
+  def check_role!(user:)
+    role_name = user[:role]
+    user[:role] = Role.find_by(name: role_name, provider: 'greenlight')
+    return if user[:role]
```

Comment:
Can we simplify this to `return if Role.exists?(...)`

Reply:
We are setting the role as `user[:role]` if found so that query is used for both setting the role and also to enable us to catch inexistent roles.

The `Role.exists` will catch the latter case but will require us to make an additional find operation to create the user.

Is that what you're suggesting Ahmad?

Reply:
Ah I see what you're doing. No this is fine